traitlets/traitlets.py

```diff
@@ -2477,8 +2477,9 @@ def instance_init(self, obj):
 class Dict(Instance):
     """An instance of a Python dict."""
     _trait = None
+    _key_trait = None
 
-    def __init__(self, trait=None, traits=None, default_value=Undefined,
+    def __init__(self, trait=None, key_trait=None, traits=None, default_value=Undefined,
```

Comment:
I'm concerned more with clarity than priority. I don't view the order as being significant (to me, there is only one positional arg for any TraitType, but Python 2 doesn't make this easy). If you'd like to make `value_trait` and `key_trait`, that would be fine by me, as long as it's done in a backward-compatible way (i.e. positional args retain their order). I wouldn't expect anyone to pass key_trait positionally ever, and I'm AOK encouraging people to pass value_trait by keyword-arg, as well, since that would be clearest.